scrunch/mutable_dataset.py

```diff
@@ -100,8 +100,9 @@ def join(self, left_var, right_ds, right_var, columns=None,
                 ]
             }
             # add the individual variable columns to the payload
-            for var in columns:
-                var_url = right_ds[var].url
+            alias_list = right_ds.resource.variables.by("alias")
+            var_urls = ['{}variables/{}/'.format(right_ds.resource.self, alias_list[alias]['id']) for alias in columns]
```

Comment:
```suggestion
            var_urls = [alias_list[alias].entity_url for alias in columns]
```

In pycrunch, we should always think it's a smell if we're interpolating strings to construct URLs, there's usually another way to get them directly from the API.